src/Views/ListView/Lists/TreeViewSetup.vala

```diff
@@ -36,29 +36,55 @@ public class Noise.TreeViewSetup : Object {
 
     public ListColumn sort_column_id { get; set; default = ListColumn.ARTIST; }
     public Gtk.SortType sort_direction { get; set; default = Gtk.SortType.ASCENDING; }
-    public ViewWrapper.Hint hint { get; set; }
     public Gda.Connection? connection { get; construct; default = null; }
     public string? uid { get; construct; default = null; }
 
     private Gee.LinkedList<Gtk.TreeViewColumn> columns = new Gee.LinkedList<Gtk.TreeViewColumn> ();
 
-    public TreeViewSetup (ViewWrapper.Hint hint, string? uid = null, Gda.Connection? connection = null) {
-        Object (
-            connection: connection,
-            hint: hint,
-            uid: uid
-        );
+    public TreeViewSetup (bool for_playlist = false, string? uid = null, Gda.Connection? connection = null) {
+        Object (uid: uid, connection: connection);
+        append_new_column (ListColumn.ICON);
```

Comment:
I'm not sure this really makes the code better. As I understand it, in an ideal world a constructor only contains calls to `Object ()` which would mean we still need something like `ViewWrapper.Hint hint` internally here even if we don't expose it externally.